pandas/core/arrays/datetimes.py

```diff
@@ -761,7 +761,10 @@ def _add_offset(self, offset) -> DatetimeArray:
         assert not isinstance(offset, Tick)
 
         if self.tz is not None:
-            values = self.tz_localize(None)
+            if not offset._use_relativedelta:
```

Comment:
is there a viable way to handle this inside the Offset code?  in theory the DTA code shouldn't need to know anything about _use_relativedelta

Reply:
The issue is with how these dates are localized, and I don't believe there is a good way to modify this in the Offset code.